### PR TITLE
feat: show dashboard cache data

### PR DESCRIPTION
**Pull request by magnew**
Closes: #7717 

### Description:

Show a message about caching on dashboards with cached charts. 

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/providers/DashboardProvider.tsx [hunk 1]
```diff
@@ -109,6 +113,7 @@ export const DashboardProvider: React.FC = ({ children }) => {
         useState<DashboardFilters>(emptyFilters);
     const [haveFiltersChanged, setHaveFiltersChanged] =
         useState<boolean>(false);
+    const [resultsCacheTimes, setResultsCacheTimes] = useState<any[]>([]);
```
Comment (ZeRego) [inline packages/frontend/src/providers/DashboardProvider.tsx:116]:
```suggestion
    const [resultsCacheTimes, setResultsCacheTimes] = useState<CacheMetadata[]>([]);
```

File: packages/frontend/src/providers/DashboardProvider.tsx [hunk 2]
```diff
@@ -429,6 +434,19 @@ export const DashboardProvider: React.FC = ({ children }) => {
...
+            setResultsCacheTimes((old) => [
+                ...old,
+                cacheMetadata.cacheUpdatedTime,
+            ]);
+        }
+    }, []);
+
+    const getOldestCacheTime = useCallback(() => {
+        return min(resultsCacheTimes);
+    }, [resultsCacheTimes]);
```
Comment (ZeRego) [inline packages/frontend/src/providers/DashboardProvider.tsx:448]: What is the advantage of returning a callback instead of the oldest cache time value?

**Review by ZeRego — APPROVED** [review]

File: packages/frontend/src/providers/DashboardProvider.tsx (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/providers/DashboardProvider.tsx:116]: Good catch

File: packages/frontend/src/providers/DashboardProvider.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/providers/DashboardProvider.tsx:448]: None really as is. I often like getters as a pattern cause they are flexible to add to, but it's not doing anything here. I can make it a value if you prefer.

**Pull request merged**